contrib/automation_tests/fragments/symbols_tab.py

```diff
@@ -72,3 +75,58 @@ def _execute(self, function_search_string):
 
         self.find_context_menu_item('Hook').click_input()
         wait_for_condition(lambda: '✓' in functions_dataview.get_item_at(0, 0).texts()[0])
+
+
+class LoadAndVerifyHelloGgpPreset(E2ETestCase):
+    """
+    Load the predefined E2E test preset and verify if has been applied correctly.
+    TODO: This may need to be updated
+    """
+    def _execute(self):
+        _show_symbols_and_functions_tabs(self.suite.top_window())
+
+        self._load_presets()
+        wait_for_condition(self._try_verify_functions_are_hooked)
+
+        Capture().execute(self.suite)
+
+        logging.info('Verifying function call counts')
+        VerifyFunctionCallCount(function_name='DrawFrame', min_calls=30, max_calls=3000).execute(self.suite)
```

Comment:
I would have expected "Capture" and "VerifyFunctionCallCount" in the "tests = [..]" list.

No need to change this now if this complicated for some reason I don't see...

Reply:
You're pointing at something that I'm not really happy with here, and I'm not sure how I want to handle this TBH. The reasoning for this is the following: I want the test case `LoadAndVerifyHelloGgpPreset` to verify the loading of the preset self-contained, and the test suite should not need to care that it needs to take a capture and count function calls to verify the loading of the preset.

I can either do this by:
- Duplicating code from `Capture` and `VerifyFunctionCallCount`, which is obviously not ideal
- Moving the functionality to capture and verify the call count into a free function in the respective modules and calling those in all test cases. This means I can't use some of the convenience that the `E2ETestCase` class provides inside these functions, and I somewhat expose internals of the `Capture` and `VerifyFunctionCallCount`
- Using the test cases directly, which results in this slightly weird calling style

I'm a bit torn between 2 and 3. I'm afraid that going the route 2 will lead to a lot of functionality being stripped out into free functions over time again, which is ok, but gives up some of the convenience of the TestCase class.

I think it's also a bit of a core question on how independent a single TestCase actually is, and I tried to find a good balance between independence and reusability. There's preconditions to some test cases, like being connected to an instance, or having a capture taken, but apart from that, I tried to have all test cases verify their success by all means necessary without the need of the test suite to know about this.

WDYT?

Reply:
I really haven't thought of a TestCase to be entirely self-contained. Is this really necessary? As you said the tests currently depend on `ConnectToStadiaInstance` being it the suite before them anyway; they don't call that themselves.

That said I'm totally fine with it the way it is - I don't think this is importent - I just wondered (and 3 looks better than 2 to me fwiw).